sphinx-src.el

```diff
@@ -53,14 +53,22 @@ LANG is a string, and the returned major mode is a symbol."
       (if (symbolp l) (symbol-name l) l))
     "-mode")))
 
+(setq sphinx-src-directive-mode-function
+  '(("code-block" . sphinx-src--get-lang-mode)
```

Comment:
``sourcecode`` is an alias for ``code-block`` (https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block)